File: LinkPasteTool_General/LinkPasteTool/Class/CElement.vb

''' <summary>
''' コンボボックスに表示用mの次列と内部IDを保持させるクラス
''' </summary>
<System.Diagnostics.DebuggerStepThrough>
Public Class CElement

    Private m_ID As String '内部ID
    Private m_Name As String    '表示用文字列

    ''' <summary>
    ''' 表示用文字列はToStringをオーバーライドして取得する
    ''' </summary>
    ''' <returns></returns>
    Public Overrides Function ToString() As String
        Return NAME
    End Function

    ''' <summary>
    ''' 実際の値
    ''' (ValueMemberに設定する文字列と同名にする)
    ''' </summary>
    ''' <returns></returns>
    Public Property ID() As String
        Get
            Return m_ID
        End Get
        Set(value As String)
            m_ID = value
        End Set
    End Property

    ''' <summary>
    ''' 表示名称
    ''' (DisplayMemberに設定する文字列と同名にする
    ''' </summary>
    ''' <returns></returns>
    Public Property NAME() As String
        Get
            Return m_Name
        End Get
        Set(value As String)
            m_Name = value
        End Set
    End Property

End Class

File: PDFしおりリンクツール/PDF_Shiori/Module/CElement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PDF_Shiori
{
	public class CElement
	{
		private string m_ID;    //内部ID
		private string m_Name;  //表示用文字列

		/// <summary>
		/// 表示用文字列はToStringをオーバーライドして取得する
		/// </summary>
		/// <returns></returns>
		public override string ToString()
		{
			return NAME;
		}

		/// <summary>
		/// 実際の値
		/// (ValueMemberに設定する文字列と同名にする)
		/// </summary>
		public string ID
		{
			get
			{
				return m_ID;
			}
			set
			{
				m_ID = value;
			}
		}

		/// <summary>
		/// 表示名称
		/// (DisplayMemberに設定する文字列と同名にする
		/// </summary>
		public string NAME
		{
			get
			{
				return m_Name;
			}
			set
			{
				m_Name = value;
			}
		}
	}


}
